libcob/common.c

```diff
@@ -5025,36 +5023,8 @@ cob_set_date_from_epoch (struct cob_time *cb_time, const unsigned char *p)
 		return 1;
 	}
 
-	/* allocate tmptr for epoch */
-	tmptr = localtime (&t);
-	/* set seconds, minutes, hours and big days */
-	tmptr->tm_sec = seconds % 60;
-	seconds /= 60;
-	tmptr->tm_min = seconds % 60;
-	seconds /= 60;
-	tmptr->tm_hour = seconds % 24;
-	seconds /= 24;
-	tmptr->tm_mday = (int)seconds;
-	tmptr->tm_isdst = -1;
-
-	/* normalize if needed (definitely for epoch, but also for example 30 Feb
-		to be changed to correct march date),
-		set tm_wday, tm_yday and tm_isdst */
-	if (mktime (tmptr) == -1) {
-		return 1;
-	}
+        set_cob_time_from_localtime ((time_t)seconds, cb_time);
```

Comment:
the differences are:

nanosecond  -1 -> 0; while I think that's better for cobc because of (cb_set_intr_when_compiled), it isn't for libcob - so please set it to -1 after set_cob_time_from_localtime here and set it to zero after calling cob_set_date_from_epoch in set_compile_date.

offset_known + utc_offset set; I think the first should be set to zero here as well

Reply:
even different - to following https://reproducible-builds.org/docs/timezones/ we should do it different than before and _always_ do `offset_known = 1; utc_offset = 0;` after calling `set_cob_time_from_localtime` here

Reply:
Made the changes in the last commit.

Reply:
thanks, that looks good to me and it should help for reproducibility and the nano=0 may fix an overflow as well